tests/unit/aiplatform/test_cloud_training_job.py

```diff
@@ -0,0 +1,224 @@
+            output_uri_prefix=_TEST_STAGING_BUCKET
+        ),
+        scheduling=gca_custom_job_compat.Scheduling(
+            timeout=duration_pb2.Duration(seconds=_TEST_TIMEOUT),
+            restart_job_on_worker_restart=_TEST_RESTART_JOB_ON_WORKER_RESTART,
+        ),
+        service_account=_TEST_SERVICE_ACCOUNT,
+        network=_TEST_NETWORK,
+    ),
+    encryption_spec=_TEST_DEFAULT_ENCRYPTION_SPEC,
+)
+
+
+@pytest.fixture
+def create_custom_job_mock():
+    with mock.patch.object(
+        job_service_client.JobServiceClient, "create_custom_job"
+    ) as create_custom_job_mock:
+        create_custom_job_mock.return_value = _get_custom_job_proto(
+            name=_TEST_CUSTOM_JOB_NAME,
+            state=gca_job_state_compat.JobState.JOB_STATE_PENDING,
+        )
+        yield create_custom_job_mock
+
+
+@pytest.fixture
+def get_custom_job_mock():
+    with patch.object(
+        job_service_client.JobServiceClient, "get_custom_job"
+    ) as get_custom_job_mock:
+        get_custom_job_mock.side_effect = [
+            _get_custom_job_proto(
+                name=_TEST_CUSTOM_JOB_NAME,
+                state=gca_job_state_compat.JobState.JOB_STATE_PENDING,
+            ),
+            _get_custom_job_proto(
+                name=_TEST_CUSTOM_JOB_NAME,
+                state=gca_job_state_compat.JobState.JOB_STATE_RUNNING,
+            ),
+            _get_custom_job_proto(
+                name=_TEST_CUSTOM_JOB_NAME,
+                state=gca_job_state_compat.JobState.JOB_STATE_SUCCEEDED,
+            ),
+        ]
+        yield get_custom_job_mock
+
+
+def _get_custom_job_proto(state=None, name=None, error=None, version="v1"):
+    custom_job_proto = copy.deepcopy(_TEST_BASE_CUSTOM_JOB_PROTO)
+    custom_job_proto.name = name
+    custom_job_proto.state = state
+    custom_job_proto.error = error
+
+    if version == "v1beta1":
+        v1beta1_custom_job_proto = gca_custom_job_v1beta1.CustomJob()
+        v1beta1_custom_job_proto._pb.MergeFromString(
+            custom_job_proto._pb.SerializeToString()
+        )
+        custom_job_proto = v1beta1_custom_job_proto
+        custom_job_proto.job_spec.tensorboard = _TEST_TENSORBOARD_NAME
+
+    return custom_job_proto
+
+
+class LinearRegression(VertexModel): 
+ 
+        # constraint on no constructor arguments
+        def __init__(self):
+            input_size = 10
+            output_size = 10
+            super(LinearRegression, self).__init__()
+            self.linear = torch.nn.Linear(input_size, output_size)
+
+        def forward(self, x):
+            return self.linear(x)
+
+        def train_loop(self, dataloader, loss_fn, optimizer):
+            size = len(dataloader.size)
+            for batch, (X, y) in enumerate(dataloader):
+                pred = self.predict(X)
+                loss = loss_fn(pred, y)
+
+                optimizer.zero_grad()
+                loss.backward()
+                optimizer.step()
+
+        def fit(self, dataset):
```

Comment:
This shouldn't take any arguments for right now because of limitations of the initial implementation.